lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -26,37 +26,49 @@
 	 ************************************/
 
 	/**
-	 * Get the presence state for this Channel. Implicitly attaches the
-	 * Channel. However, if the channel is in or moves to the FAILED
-	 * state before the operation succeeds, it will result in an error
-	 * @return: the current present members.
+	 * String parameter names to get() call with Param[] as an argument
+	 */
+	public final static String GET_WAITFORSYNC = "waitForSync";
+	public final static String GET_CLIENTID = "clientId";
+	public final static String GET_CONNECTIONID = "connectionId";
+
+	/**
+	 * Get the presence state for this channel. Take Param[] array as an argument
+	 * @param params
+	 * @return
 	 * @throws AblyException
+	 * @throws InterruptedException
 	 */
-	public synchronized PresenceMessage[] get() throws AblyException {
+	public synchronized PresenceMessage[] get(Param... params) throws InterruptedException, AblyException {
```

Comment:
`InterruptedException` is IMHO semantically correct, it is a result of client calling 'Thread.interrupt()`. You don't have an `InterruptedException` in HTTP code because you have to work with `HttpUrlConnection` and it has a lot of shortcomings.
Nonetheless, I changed it to `AblyException.fromThrowable(e)`. It breaks code too. First, the default for `Presence.get()` is now to wait for SYNC to complete. And secondly, several versions of `Presence.get()` are no longer throwing `InterruptedException`